src/pykeen/models/predict.py

```diff
@@ -727,3 +731,230 @@ def predict_triples_df(
+        runtime.
+
+    :return:
+        A tuple (score_mean, score_std) of the mean and std of the scores sampled from the dropout distribution.
+        The std may be interpreted as a measure of uncertainty.
+    """
+    dropout_modules = get_dropout_modules(model)
+    if not dropout_modules:
+        raise ValueError(
+            "Model needs to contain at least one dropout layer to use the Monte-Carlo Dropout technique.",
+        )
+
+    # Enforce evaluation mode
+    model.eval()
+
+    # set dropout layers to training mode
+    for module in dropout_modules:
+        module.train()
+
+    kwargs = dict()
+    if slice_size is not None:
+        kwargs["slice_size"] = slice_size
+
+    # draw samples
+    batch = batch.to(model.device)
+    scores = torch.stack([score_method(batch, **kwargs) for _ in range(num_samples)], dim=0)
+    if model.predict_with_sigmoid:
+        scores = torch.sigmoid(scores)
+
+    # compute mean and std
+    return UncertainPrediction(score=scores.mean(dim=0), uncertainty=scores.std(dim=0))
+
+
+def predict_hrt_uncertain(
+    model: Model,
+    hrt_batch: torch.LongTensor,
+    num_samples: int = 5,
+) -> UncertainPrediction:
+    """
+    Calculate the scores for triples and add an uncertainty quantification via Monto-Carlo dropout.
+
+    .. seealso ::
+        score_hrt, _predict_uncertain
+
+    .. note ::
+        this method requires the model to have at least one dropout layer.
+
+    :param model:
+        the model used for predicting scores
+    :param hrt_batch: shape: (number of triples, 3), dtype: long
+        The indices of (head, relation, tail) triples.
+    :param num_samples: >1
+        the number of samples to draw
+
+    :return: shape: (number of triples, 1), dtype: float
+        The score for each triple, and an uncertainty score, where larger scores correspond to less certain
+        predictions.
+    """
+    return _predict_uncertain(
+        model=model,
+        batch=hrt_batch,
+        score_method=model.score_hrt,
+        num_samples=num_samples,
+    )
+
+
+def predict_h_uncertain(
+    model: Model,
+    rt_batch: torch.LongTensor,
+    num_samples: int = 5,
+    slice_size: Optional[int] = None,
+) -> UncertainPrediction:
+    """Forward pass using left side (head) prediction for obtaining scores of all possible heads.
+
+    This method calculates the score for all possible heads for each (relation, tail) pair, as well as an uncertainty
+    quantification.
+
+    Additionally, the model is set to evaluation mode.
+
+    .. note::
+
+        If the model has been trained with inverse relations, the task of predicting
+        the head entities becomes the task of predicting the tail entities of the
+        inverse triples, i.e., $f(*,r,t)$ is predicted by means of $f(t,r_{inv},*)$.
+
+    .. note ::
+        this method requires the model to have at least one dropout layer.
```

Comment:
here we have quite a bit of copy-paste from `score_h`

Reply:
If it's mostly copied, why not just write the important parts and link to `score_h` for the rest?